design-proposals/klip-48-schemas.md

```diff
@@ -0,0 +1,115 @@
+
+In addition to providing a means to logically organize groups of database objects, schemas also provide basic operational benefits. The most obvious of these benefits is the ability to drop an entire schema in a manner that recursively drops all objects underneath that schema. It is easy to imagine additional benefits that schemas have the potential to provide for ksqlDB users longer term: performing basic operational tasks such as starting and stopping groups of persistent queries, applying configuration properties to an entire schema of objects, and more.
+
+In this KLIP we will propose the initial introduction of schemas into ksqlDB’s grammar, and describe the related functionality that this grammar invokes.
+
+## Scope
+
+### What is in scope
+
+* The overall user experience of ksqlDB schemas will be proposed.
+* The minimally necessary grammar required to introduce schemas will be proposed.
+* New configuration relating to schemas will be proposed.
+
+### What is not in scope
+
+* The proposed grammar is not exhaustive. It does not include all potential parameters or optional grammar constructs that we may want to support for creating and/or interacting with schemas.
+* No implementation details will be proposed.
+
+
+## Value/return
+
+The most immediate benefit of introducing schemas into ksqlDB is that they will provide a means to better organize workloads in a way that is familiar to users who have experience using other SQL-based systems. Furthermore, as ksqlDB continues to gain widespread adoption, users are increasingly leveraging it to run expansive, complex workloads that would benefit from having more organizational hierarchy.
+
+Additionally, the introduction of a schema abstraction will lay some of the groundwork for enabling users to perform certain operational tasks over groups of database objects in the future. For example, having the ability to [stop and start](https://github.com/confluentinc/ksql/issues/6403) subsets of persistent queries is something we know users want. Schemas have the potential to provide a mechanism for performing such operations efficiently.
+
+## Public APIs
+
+ksqlDB schemas will be primarily leveraged via SQL syntax, with some optional configuration. Each of the new syntax constructs associated with ksqlDB schemas are proposed as follows:
+
+### Creating schemas
+
+```sql
+CREATE SCHEMA my_schema;
+```
+
+This statement will create a new, empty schema. Once a schema is created, it may be used to qualify any newly created objects:
```

Comment:
From my understanding, in most systems it is not possible to create an object outside (ie, without) a schema, and if no schema is specified is a CREATE, the default schema is use. I also believe (not 100%) that the default schema cannot be dropped?

This part seems important with regard to upgrading, because it would imply that all existing objects would be migrated into the default schema.